pkg/postgres/walker/schema.go

```diff
@@ -157,8 +158,28 @@ func (s *Schema) SetOptionsMap(optionsMap search.OptionsMap) {
 
 // SetSearchScope sets search scope for the schema.
 func (s *Schema) SetSearchScope(searchCategories ...v1.SearchCategory) {
+	// TODO(ROX-28123): Clean up
+	normalizedSkipMap := set.NewStringSet(
+		v1.SearchCategory_IMAGE_VULNERABILITIES.String(),
+		v1.SearchCategory_COMPONENT_VULN_EDGE.String(),
+		v1.SearchCategory_IMAGE_COMPONENTS.String(),
+		v1.SearchCategory_IMAGE_COMPONENT_EDGE.String(),
+		v1.SearchCategory_IMAGE_VULN_EDGE.String())
+
+	flattenedSkipMap := set.NewStringSet(
+		v1.SearchCategory_IMAGE_VULNERABILITIES_V2.String(),
+		v1.SearchCategory_IMAGE_COMPONENTS_V2.String())
+
 	s.SearchScope = make(map[v1.SearchCategory]struct{})
 	for _, cat := range searchCategories {
+		// The flattened CVE schema and the original interfere with each other.  We only want
+		// to register the proper search tags depending upon the feature flag.
+		if features.FlattenCVEData.Enabled() && normalizedSkipMap.Contains(cat.String()) {
```

Comment:
Is this added to prevent the sql query builder from joining old tables with new ones?

Reply:
yeah, the general idea is when it navigates the scopes to find what can be seen, it won't find these.